src/TurbulenceClosures/vertically_implicit_diffusion_solver.jl

```diff
@@ -123,13 +125,19 @@ function implicit_diffusion_solver(::VerticallyImplicitTimeDiscretization, grid)
                                  "grids that are Bounded in the z-direction.")
 
     z_solver = BatchedTridiagonalSolver(grid;
-                                        lower_diagonal = maybe_tupled_ivd_lower_diagonal,
-                                        diagonal = ivd_diagonal,
-                                        upper_diagonal = maybe_tupled_ivd_upper_diagonal)
+                                        lower_diagonal = Val(:maybe_tupled_ivd_lower_diagonal),
+                                        diagonal       = Val(:ivd_diagonal),
+                                        upper_diagonal = Val(:maybe_tupled_ivd_upper_diagonal))
```

Comment:
This seems complicated. Can we come up with a better design?

Reply:
I’ll raise an issue?

the thing is that we can’t pass a function as an argument to a kernel.

Reply:
That can't quite be true because we pass `forcing` into the tendency kernels.

Still, there's no reason why we have to use the "function" interface here. There's many possible solutions: array-like structs, new types + extending `get_coefficient` (probably the best solution), or redesigning the solver. It's not like there's only one solution.

Reply:
maybe I am wrong but new types + extending `get_coefficient` sounds a bit like this with types instead of `Val{:function_name}`). I guess probably redesigning the solvers might be a long-term solution that we can explore